client/src/modules/stock/exit/exit.js

```diff
@@ -318,10 +325,43 @@ function StockExitController(
     StockModal.openFindPatient({ entity_uuid : vm.selectedEntityUuid })
       .then(patient => {
         handleSelectedEntity(patient, 'patient');
+        loadInvoiceInventories(patient);
       })
       .catch(Notify.handleError);
   }
 
+  // load inventories from an invoice
+  function loadInvoiceInventories(patient) {
+    vm.inventoryNotAvailable = [];
+    vm.stockForm.clear();
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.EDIT);
+
+    if (!patient || !patient.invoice) { return; }
+
+    const nbItems = patient.invoice.items.length;
+
+    if (!nbItems) { return; }
+
+    let inventoryAvailableIndex = 0;
+
+    patient.invoice.items.forEach((item) => {
+      if (vm.mapSelectableInventories[item.inventory_uuid]) {
+        addItems(1);
+        vm.stockForm.store.data[inventoryAvailableIndex].inventory = vm.mapSelectableInventories[item.inventory_uuid];
```

Comment:
Hmm..  it looks like you need to have the last item inserted into the store.  The [PatientInvoiceFormService](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/invoices/patientInvoiceForm.service.js#L375) deals with this by simply returning the last element inserted into the store. I propose we do the same thing here.

Modify [the StockFormService](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/stock/StockForm.service.js#L52) to `return elt;` as the last line.  Then you can write this code, which will be easier to read:
```js
patient.invoice.items.forEach((item) => {
  if (vm.mapSelectableInventories[item.inventory_uuid]) {
    const row = vm.stockForm.addItems(1);
    row.inventory = vm.mapSelecteableInventories[item.inventory_uuid];
    row.inventory_uuid = item.inventory_uuid;
    // etc..
  }
});